Broken tests: RazorCompile_NoopsWithNoFiles et all

These tests [fail](http://aspnetci/viewLog.html?buildId=456519&tab=buildResultsDiv&buildTypeId=XPlat_HighSierraOsxUniverse) on Universe test runs:

```
Microsoft.AspNetCore.Razor.Design.IntegrationTests.Assert+BuildFailedException : Build failed.
dotnet msbuild /nr:false /restore /p:_RazorBuildServerPipeName=0dae5e52-a070-466f-841a-82e90065d7d6 /t:RazorCompile /p:Configuration=Release 

Microsoft (R) Build Engine version 15.7.172.38367 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restoring packages for /Users/aspnetci/Documents/buildAgent/temp/buildTmp/Razor/cf21ufjh.qaw/SimpleMvc/SimpleMvc.csproj...
/Users/aspnetci/Documents/buildAgent/temp/buildTmp/Razor/cf21ufjh.qaw/SimpleMvc/SimpleMvc.csproj : warning NU1603: SimpleMvc depends on Microsoft.NETCore.App (>= 2.1.0-rc1-26424-03) but Microsoft.NETCore.App 2.1.0-rc1-26424-03 was not found. An approximate best match of Microsoft.NETCore.App 2.2.0-preview1-26424-04 was resolved.
  Generating MSBuild file /Users/aspnetci/Documents/buildAgent/temp/buildTmp/Razor/cf21ufjh.qaw/SimpleMvc/obj/SimpleMvc.csproj.nuget.g.props.
  Generating MSBuild file /Users/aspnetci/Documents/buildAgent/temp/buildTmp/Razor/cf21ufjh.qaw/SimpleMvc/obj/SimpleMvc.csproj.nuget.g.targets.
  Restore completed in 975.74 ms for /Users/aspnetci/Documents/buildAgent/temp/buildTmp/Razor/cf21ufjh.qaw/SimpleMvc/SimpleMvc.csproj.
/Users/aspnetci/Documents/buildAgent/temp/buildTmp/Razor/cf21ufjh.qaw/SimpleMvc/SimpleMvc.csproj : warning NU1603: SimpleMvc depends on Microsoft.NETCore.App (>= 2.1.0-rc1-26424-03) but Microsoft.NETCore.App 2.1.0-rc1-26424-03 was not found. An approximate best match of Microsoft.NETCore.App 2.2.0-preview1-26424-04 was resolved.
/Users/aspnetci/Documents/buildAgent/temp/buildTmp/Razor/cf21ufjh.qaw/SimpleMvc/SimpleMvc.csproj : error : The project was restored using Microsoft.NETCore.App version 2.2.0-preview1-26424-04, but with current settings, version 2.1.0-rc1-26424-03 would be used instead.  To resolve this issue, make sure the same settings are used for restore and for subsequent operations such as build or publish.  Typically this issue can occur if the RuntimeIdentifier property is set during build or publish but not during restore.



   at Microsoft.AspNetCore.Razor.Design.IntegrationTests.Assert.BuildPassed(MSBuildResult result) in /_/test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/Assert.cs:line 27
   at Microsoft.AspNetCore.Razor.Design.IntegrationTests.RazorCompileIntegrationTest.RazorCompile_NoopsWithNoFiles() in /_/test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/RazorCompileIntegrationTest.cs:line 45
--- End of stack trace from previous location where exception was thrown ---
```

This is almost certainly happening because we updated the SDK. If we set `/p:VerifyMatchingImplicitPackageVersion=false` we get:

```
System.TypeInitializationException : The type initializer for 'Microsoft.CodeAnalysis.Razor.Workspaces.DefaultTagHelperDescriptorFactoryTest' threw an exception.
---- System.InvalidOperationException : Cannot find reference assembly 'Microsoft.AspNetCore.Html.Abstractions.dll' file for package Microsoft.AspNetCore.Html.Abstractions.Reference
Stack Trace:
at Microsoft.CodeAnalysis.Razor.Workspaces.DefaultTagHelperDescriptorFactoryTest.CreateDescriptor_BuildsDescriptorsFromSimpleTypes() in D:\dd\Razor\test\Microsoft.CodeAnalysis.Razor.Test\DefaultTagHelperDescriptorFactoryTest.cs:line 1756
----- Inner Stack Trace -----
at Microsoft.Extensions.DependencyModel.Resolution.ReferenceAssemblyPathResolver.TryResolveAssemblyPaths(CompilationLibrary library, List`1 assemblies)
at Microsoft.Extensions.DependencyModel.Resolution.CompositeCompilationAssemblyResolver.TryResolveAssemblyPaths(CompilationLibrary library, List`1 assemblies)
at Microsoft.Extensions.DependencyModel.CompilationLibrary.ResolveReferencePaths(ICompilationAssemblyResolver resolver, List`1 assemblies)
at System.Linq.Enumerable.SelectManySingleSelectorIterator`2.MoveNext()
at System.Linq.Enumerable.SelectEnumerableIterator`2.ToArray()
at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
at Microsoft.CodeAnalysis.TestCompilation.GetMetadataReferences(Assembly assembly) in D:\dd\Razor\test\Microsoft.AspNetCore.Razor.Test.Common\TestCompilation.cs:line 30
at Microsoft.CodeAnalysis.TestCompilation.Create(Assembly assembly, SyntaxTree syntaxTree) in D:\dd\Razor\test\Microsoft.AspNetCore.Razor.Test.Common\TestCompilation.cs:line 41
at Microsoft.CodeAnalysis.Razor.Workspaces.DefaultTagHelperDescriptorFactoryTest..cctor() in D:\dd\Razor\test\Microsoft.CodeAnalysis.Razor.Test\DefaultTagHelperDescriptorFactoryTest.cs:line 25
```
instead.

CC @pranavkm  who said he'd look into this.

--- a/build/repo.targets
+++ b/build/repo.targets
@@ -24,9 +24,8 @@
     <PropertyGroup>
       <TemplateProperties>
         MSBuildLocation=$(VisualStudioMSBuildx86Path);
-        MicrosoftNETCoreAppVersion=$(MicrosoftNETCoreApp22PackageVersion);
-        MicrosoftNETCoreApp22PackageVersion=$(MicrosoftNETCoreApp22PackageVersion);
-        NETStandardLibraryVersion=$(NETStandardLibrary20PackageVersion)
+        MicrosoftNETCoreAppPackageVersion=$(MicrosoftNETCoreApp22PackageVersion);
+        NETStandardLibraryPackageVersion=$(NETStandardLibrary20PackageVersion)
       </TemplateProperties>
     </PropertyGroup>
 

--- a/test/Microsoft.AspNetCore.Razor.Design.Test/BuildVariables.cs
+++ b/test/Microsoft.AspNetCore.Razor.Design.Test/BuildVariables.cs
@@ -6,6 +6,8 @@ namespace Microsoft.AspNetCore.Razor.Design.IntegrationTests
     internal static partial class BuildVariables
     {
         private static string _msBuildPath = string.Empty;
+        private static string _microsoftNETCoreAppPackageVersion = string.Empty;
+        private static string _netStandardLibraryPackageVersion = string.Empty;
 
         static partial void InitializeVariables();
 
@@ -15,6 +17,24 @@ namespace Microsoft.AspNetCore.Razor.Design.IntegrationTests
             {
                 InitializeVariables();
                 return _msBuildPath;
+            }
+        }
+
+        public static string MicrosoftNETCoreAppPackageVersion
+        {
+            get
+            {
+                InitializeVariables();
+                return _microsoftNETCoreAppPackageVersion;
+            }
+        }
+
+        public static string NETStandardLibraryPackageVersion
+        {
+            get
+            {
+                InitializeVariables();
+                return _netStandardLibraryPackageVersion;
             }
         }
     }

--- a/test/Microsoft.AspNetCore.Razor.Design.Test/BuildVariables.cs.template
+++ b/test/Microsoft.AspNetCore.Razor.Design.Test/BuildVariables.cs.template
@@ -8,6 +8,8 @@ namespace Microsoft.AspNetCore.Razor.Design.IntegrationTests
         static partial void InitializeVariables()
         {
             _msBuildPath = @"${MSBuildLocation}";
+            _microsoftNETCoreAppPackageVersion = "${MicrosoftNETCoreAppPackageVersion}";
+            _netStandardLibraryPackageVersion = "${NETStandardLibraryPackageVersion}";
         }
     }
 }

--- a/test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/MSBuildIntegrationTestBase.cs
+++ b/test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/MSBuildIntegrationTestBase.cs
@@ -70,6 +70,9 @@ namespace Microsoft.AspNetCore.Razor.Design.IntegrationTests
 
                 // Let the test app know it is running as part of a test.
                 "/p:RunningAsTest=true",
+
+                $"/p:MicrosoftNETCoreAppVersion={BuildVariables.MicrosoftNETCoreAppPackageVersion}",
+                $"/p:NETStandardLibraryPackageVersion={BuildVariables.NETStandardLibraryPackageVersion}",
             };
 
             if (!suppressRestore)

--- a/test/testapps/Directory.Build.props
+++ b/test/testapps/Directory.Build.props
@@ -10,13 +10,14 @@
     <RazorSdkCurrentVersionTargets>$(SolutionRoot)src\Microsoft.NET.Sdk.Razor\build\netstandard2.0\Sdk.Razor.CurrentVersion.targets</RazorSdkCurrentVersionTargets>
   </PropertyGroup>
 
-  <Import Project="$(SolutionRoot)build\dependencies.props" />
+  <!-- Import solution dependencies.props when building in place -->
+  <Import Project="..\..\build\dependencies.props" Condition="'$(BinariesRoot)'==''"/>
 
   <PropertyGroup>
     <!-- aspnet/BuildTools#662 Don't police what version of NetCoreApp we use -->
     <NETCoreAppMaximumVersion>99.9</NETCoreAppMaximumVersion>
 
-    <NETStandardImplicitPackageVersion>$(NETStandardLibrary20PackageVersion)</NETStandardImplicitPackageVersion>
+    <NETStandardImplicitPackageVersion>$(NETStandardLibraryPackageVersion)</NETStandardImplicitPackageVersion>
   </PropertyGroup>
 
   <!-- Don't use the server when building in place. This locks up rzc.dll -->

--- a/test/testapps/Directory.Build.targets
+++ b/test/testapps/Directory.Build.targets
@@ -2,7 +2,7 @@
   <Import Project="RazorTest.Introspection.targets" />
 
   <PropertyGroup>
-    <RuntimeFrameworkVersion Condition=" '$(TargetFramework)' == 'netcoreapp2.2' ">$(MicrosoftNETCoreApp22PackageVersion)</RuntimeFrameworkVersion>
+    <RuntimeFrameworkVersion Condition=" '$(TargetFramework)' == 'netcoreapp2.2' ">$(MicrosoftNETCoreAppVersion)</RuntimeFrameworkVersion>
     <!-- aspnet/BuildTools#662 Don't police what version of NetCoreApp we use -->
     <NETCoreAppMaximumVersion>99.9</NETCoreAppMaximumVersion>
   </PropertyGroup>

